README.md

```diff
@@ -5,11 +5,11 @@
 
 Geode Loopless (GeodeL) is a Domain Specific Language that aims at facilitating the writing and analysis of Primitive Recursive Functions and Sets.
 
-This DSL is embedded in Scala, and aims to use it's strong type system to enforce the arity and type of the PrimRec functions and sets, thus ensuring any function or set is really primitive recursive.
+This DSL is embedded in Scala, and aims to use its strong type system to enforce the arity and type of the `PrimRec` functions and sets, thus ensuring any function or set is really primitive recursive.
 
-Primitive recursive functions are all the functions created by combining the successor function and constants with projection, recursion and composition.
+Primitive recursive functions are all the functions that can be created by combining the successor function on naturals and constants with projection, recursion, and composition.
 
-In Scala this translates to the following ([the actual implementation](https://github.com/Sporarum/GeodeLoopless/blob/main/src/main/scala/PrimRecFun.scala) also includes a "stack trace" method `debug`):
+In Scala, this translates to the following ([the actual implementation](https://github.com/Sporarum/GeodeLoopless/blob/main/src/main/scala/PrimRecFun.scala) also includes a "stack trace" method `debug`):
```

Comment:
```suggestion
In Scala, this translates to the following ([the actual implementation](src/main/scala/PrimRecFun.scala) also includes a "stack trace" method `debug`):
```
We can do it here as well, no ?

Reply:
This was the one I suggested (line 12). I did not do it myself because I thought I'd get your opinion on it. It's a somewhat subjective choice.

Reply:
Yeah I see, I think it makes sense for this one, but not for the things in `DerivedFunctions`